features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Create/Default/APICreateDefault.jsx

```
@@ -198,45 +201,112 @@ function APICreateDefault(props) {
+                        .then(() => {
+                            Alert.info('API Revision Deployed Successfully');
+                            setIsDeploying(false);
+                        })
+                        .catch((error) => {
+                            if (error.response) {
+                                Alert.error(error.response.body.description);
+                                setPageError(error.response.body);
+                            } else {
+                                const message = 'Something went wrong while deploying the API Revision';
+                                Alert.error(intl.formatMessage({
+                                    id: 'Apis.Create.Default.APICreateDefault.error.errorMessage.deploy.revision',
+                                    defaultMessage: message,
+                                }));
+                                setPageError(message);
+                            }
+                            console.error(error);
+                        })
+                        .finally(() => {
+                            setIsDeploying(false);
+                        });
+                })
+                .catch((error) => {
+                    if (error.response) {
+                        Alert.error(error.response.body.description);
+                        setPageError(error.response.body);
+                    } else {
+                        const message = 'Something went wrong while creating the API Revision';
+                        Alert.error(intl.formatMessage({
+                            id: 'Apis.Create.Default.APICreateDefault.error.errorMessage.create.revision',
+                            defaultMessage: message,
+                        }));
+                        setPageError(message);
+                    }
+                    console.error(error);
+                })
+                .finally(() => {
+                    setIsRevisioning(false);
```

I think we should throw an exception here and break the execution. Prevent it from publishing and deploying